src/Butler/App/TodoManager.hs

```diff
@@ -0,0 +1,149 @@
+                    input_
+                        ( [type_ "checkbox"]
+                            <> case taskSelected of
+                                TaskSelected -> [checked_]
+                                TaskNotSelected -> mempty
+                        )
+
+                div_ [] $ toHtml taskDesc
+
+addTask :: TodoManager -> Text -> TodoManager
+addTask (TodoManager (TaskIndex i) todoTasks) content =
+    let newId = i + 1
+        task = newTask content (TaskId newId)
+     in TodoManager (TaskIndex newId) (task : todoTasks)
+
+delSelectedTasks :: TodoManager -> TodoManager
+delSelectedTasks (TodoManager taskIndex todoTasks) =
+    TodoManager taskIndex $
+        filter
+            ( \(TodoTask _ taskSelected _) ->
+                case taskSelected of
+                    TaskSelected -> False
+                    TaskNotSelected -> True
+            )
+            todoTasks
+
+selectTask :: TodoTask -> TodoTask
+selectTask (TodoTask taskId taskSelected taskDesc) =
+    let newSelectState = case taskSelected of
+            TaskSelected -> TaskNotSelected
+            TaskNotSelected -> TaskSelected
+     in TodoTask taskId newSelectState taskDesc
+
+newTask :: Text -> TaskId -> TodoTask
+newTask content taskId = TodoTask taskId TaskNotSelected $ TaskDesc content
+
+setSelectedTask :: TodoManager -> TaskId -> TodoManager
+setSelectedTask (TodoManager taskIndex todoTasks) taskId =
+    TodoManager taskIndex $
+        map
+            ( \task@(TodoTask tId _ _) ->
+                if tId == taskId
+                    then selectTask task
+                    else task
+            )
+            todoTasks
+
+startTodoManager :: AppContext -> ProcessIO ()
+startTodoManager ctx = do
+    logInfo "TodoManager started!" []
+    let appState = TodoManager 0 []
+        memAddr = "todo-manager-" <> showT ctx.wid <> ".bin"
+    (_, appStateM) <- newProcessMemory (from memAddr) (pure appState)
+    let mountUI = with div_ [wid_ ctx.wid "w"] $ appUI ctx appStateM
+
+    -- Handle events
+    forever do
+        atomically (readPipe ctx.pipe) >>= \case
+            AppDisplay (UserJoined client) -> atomically $ sendHtml client mountUI
+            AppTrigger ev -> do
+                case ev.trigger of
+                    "add-item" -> do
+                        logInfo "Adding item" ["ev" .= ev]
+                        case ev.body ^? key "taskDesc" . _JSON of
+                            Just taskDesc -> do
+                                atomically $ modifyMemoryVar appStateM $ \tm -> do
+                                    addTask tm taskDesc
+                            Nothing -> pure ()
+                    "del-item" -> do
+                        logInfo "Removing item" ["ev" .= ev]
+                        atomically $ modifyMemoryVar appStateM $ \tm -> do
+                            delSelectedTasks tm
+                    "checkbox-click" -> do
```

Comment:
There is a similar checkbox handler in the [FileManager](https://github.com/ButlerOS/haskell-butler/blob/main/src/Butler/App/FileManager.hs) (search for selected). It seems like we could factor the code to make such feature re-usable.